### PR TITLE
Add Custom Geth Health Check Wait For RPC Strategy

### PR DESCRIPTION
**Pull request by tateexon**
also fixes linter and nancy errors and some sonar code smells

### CODE REVIEW

**Review by kalverra — APPROVED** [review]

File: docker/test_env/geth.go [hunk 1]
```diff
@@ -164,15 +169,16 @@ func (g *Geth) getGethContainerRequest(networks []string) (*tc.ContainerRequest,
...
 		Networks:        networks,
 		WaitingFor: tcwait.ForAll(
 			tcwait.NewHTTPStrategy("/").
-				WithPort("8544/tcp"),
+				WithPort(exposedHttpPort),
 			tcwait.ForLog("WebSocket enabled"),
 			tcwait.ForLog("Started P2P networking").
 				WithStartupTimeout(120*time.Second).
 				WithPollInterval(1*time.Second),
+			NewWebSocketStrategy(),
```
Comment (AnieeG) [inline docker/test_env/geth.go:181]: please add this similar strategy for non-dev-geth as well

File: docker/test_env/geth_test.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+	"github.com/stretchr/testify/require"
+
+	"github.com/smartcontractkit/chainlink-testing-framework/docker"
+)
+
+func TestGeth(t *testing.T) {
+	network, err := docker.CreateNetwork()
+	require.NoError(t, err)
+	g := NewGeth([]string{network.Name})
+	_, _, err = g.StartContainer()
```
Comment (AnieeG) [inline docker/test_env/geth_test.go:15]: It's worth adding the creation of a blockchain.EVMClient here

File: docker/test_env/geth.go [hunk 3]
```diff
@@ -71,20 +76,20 @@ func (g *Geth) StartContainer() (blockchain.EVMNetwork, InternalDockerUrls, erro
...
-	wsPort, err := ct.MappedPort(context.Background(), "8545/tcp")
+	wsPort, err := ct.MappedPort(context.Background(), exposedWsPort)
 	if err != nil {
 		return blockchain.EVMNetwork{}, InternalDockerUrls{}, err
 	}
 
 	g.Container = ct
 	g.ExternalHttpUrl = fmt.Sprintf("http://%s:%s", host, httpPort.Port())
-	g.InternalHttpUrl = fmt.Sprintf("http://%s:8544", g.ContainerName)
+	g.InternalHttpUrl = fmt.Sprintf("http://%s:%s", g.ContainerName, httpPort)
```
Comment (AnieeG) [inline docker/test_env/geth.go:90]:
```suggestion
	g.InternalHttpUrl = fmt.Sprintf("http://%s:%s", g.ContainerName, httpPort.Port())
```

Comment (tateexon) [reply docker/test_env/geth.go]: This is actually wrong. I had a constant "httpPort" that was getting overridden 🤦 Fixed now

File: docker/test_env/geth.go [hunk 4]
```diff
@@ -71,20 +76,20 @@ func (g *Geth) StartContainer() (blockchain.EVMNetwork, InternalDockerUrls, erro
...
 	g.ExternalWsUrl = fmt.Sprintf("ws://%s:%s", host, wsPort.Port())
-	g.InternalWsUrl = fmt.Sprintf("ws://%s:8545", g.ContainerName)
+	g.InternalWsUrl = fmt.Sprintf("ws://%s:%s", g.ContainerName, wsPort)
```
[the remark this answers is not in the document]
Comment (tateexon) [reply docker/test_env/geth.go]: Same as above

File: docker/test_env/geth_test.go (hunk 2 above)
Comment (tateexon) [reply docker/test_env/geth_test.go:16]: Good call, added

File: docker/test_env/geth.go (hunk 1 above)
Comment (tateexon) [reply docker/test_env/geth.go]: Also a good call, done.

**Review by AnieeG — APPROVED** [review]
LGTM

**Pull request merged**